DBZ-866 Add schema, database and table to Postgres connector

oripwk opened the pull request:
https://issues.jboss.org/projects/DBZ/issues/DBZ-866

gunnarmorling commented on debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java:
```diff
@@ -546,6 +546,8 @@ protected void assertRecordOffset(SourceRecord record, boolean expectSnapshot, b
         assertNotNull(offset.get(SourceInfo.TXID_KEY));
         assertNotNull(offset.get(SourceInfo.TIMESTAMP_KEY));
         assertNotNull(offset.get(SourceInfo.LSN_KEY));
+        assertNotNull(offset.get(SourceInfo.SCHEMA_NAME_KEY));
```
How about actually asserting the DB, schema and table names here? It should be possible to pass a `TableId` with the expected attributes to this method.

gunnarmorling commented on debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/SourceInfo.java:
```diff
@@ -116,6 +127,8 @@ protected void load(Map<String, Object> lastStoredOffset) {
         this.txId = ((Number) lastStoredOffset.get(TXID_KEY)).longValue();
         this.useconds = (Long) lastStoredOffset.get(TIMESTAMP_KEY);
         this.snapshot = lastStoredOffset.containsKey(SNAPSHOT_KEY);
+        this.schemaName = (String) lastStoredOffset.get(SCHEMA_NAME_KEY);
```
I don't think this info should be part of the *offset* as passed to Kafka Connect, only it should be added to the `source` `Struct`. Unfortunately, the class here is mangling the two aspects a bit, so I see how it can be confusing. Also see [DBZ-797](https://issues.jboss.org/browse/DBZ-797) which is about a similar case, i.e. a field is added to the "source" block although it's only meant for the offset.

The design is clearer for the new Oracle and SQL Server connectors, and eventually we should do the same for Postgres and the other existing connectors. No need to do this now as part of this change here, though, just let's not add more feels to the offset that are not needed there.

gunnarmorling commented on debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/SourceInfo.java:
```diff
@@ -150,6 +163,12 @@ protected void load(Map<String, Object> lastStoredOffset) {
         if (lsn != null) {
             result.put(LSN_KEY, lsn);
         }
+        if (schemaName != null) {
```
The same comment as above applies to here.

gunnarmorling commented on debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/SourceInfo.java:
```diff
@@ -199,15 +231,16 @@ protected Schema schema() {
      * @return the source partition and offset {@link Struct}; never null
      * @see #schema()
      */
     protected Struct source() {
```
Taking the remark about the offset from above into account, the new fields must be explicitly added here.

gunnarmorling commented:
Thanks a lot, @oripwk! I've added a few remarks inline. Once they are addressed, this can be integrated. Thanks again!

oripwk commented:
* Removed `schemaName` and `tableName` from `SourceInfo`'s offset
* Changed the tests so that DB name, schema name and table name are verified spearately

gunnarmorling commented:
Rebased and applied. Thanks a lot, @oripwk!

gunnarmorling commented:
@oripwk One more thing, can you also file a PR for updating the [PG connector docs](https://github.com/debezium/debezium.github.io/blob/develop/docs/connectors/postgresql.asciidoc); we show the source block in some examples and I think there's also a description of its elements. Thanks!

oripwk commented:
https://github.com/debezium/debezium.github.io/pull/193
